common/src/os.rs

```diff
@@ -21,3 +21,19 @@ pub fn errno() -> io::Error {
 pub fn errno() -> io::Error {
     io::Error::last_os_error()
 }
+
+#[cfg(unix)]
+pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
+    use std::os::unix::ffi::OsStrExt;
+    Ok(std::ffi::OsStr::from_bytes(b))
+}
+
+#[cfg(not(unix))]
+pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
+    Ok(std::str::from_utf8(b)?.as_ref())
+}
```

Comment:
Does this work?
```suggestion
pub fn bytes_as_osstr(b: &[u8]) -> Result<&std::ffi::OsStr, Utf8Error> {
    #[cfg(unix)]
    {
        use std::os::unix::ffi::OsStrExt;
        Ok(std::ffi::OsStr::from_bytes(b))
    }
    #[cfg(not(unix))]
    {
        Ok(std::str::from_utf8(b)?.as_ref())
    }
}
```

Reply:
I prefer to keep separated functions. Those functions are small enough. But when the logic goes longer, single function commonly leads to `#[cfg]` spaghetti here and their while separated functions leads to create a new shared function.